### ISSUE TITLE
How to use labels in training + how to use CLIP guidance in inferrence

### ISSUE THREAD

**Issue by lalalune** [nv-tlabs/GET3D#34]
In the paper this is mentioned, however it is not clear how to get this working.

**Comment by SteveJunGao** [comment]
Hi @lalalune,

Re: use labels in training:

Can you provide more details on what you mean use labels in training? we didn't mention it in the paper about using labels to my understanding. If you want to add the using labels into this codebase, I guess the code should support it as well, but it might also have some problems as I haven't tested this part. You can look at this [codebase](https://github.com/NVlabs/stylegan3), our project is based on Stylegan3 codebase and it should be able to support using labels in training. 


Re: use CLIP guidance in inference:

As mentioned in another [issue](https://github.com/nv-tlabs/GET3D/issues/20), we haven't released this part yet. If you urgently need that part I recommend you go to this [codebase](https://github.com/rinongal/StyleGAN-nada), our implementation for CLIP guidance is based on this code.

**Comment by SteveJunGao** [comment]
Ah, I see,

Sorry for the confusion! The "source text label" here doesn't mean the input to our network, it means the text prompt to compute the directional CLIP loss proposed in the [StyleGAN-NADA](https://stylegan-nada.github.io/) to fine-tune the generator. 

In the code, `use_labels` and `has_labels` only intend to support training with datasets that have labels, e.g. training with Shapenet 13 categories, you can add labels to indicate which category the image is, although we haven't used it in our paper and code.

**Comment by SteveJunGao** [comment]
Hi @lalalune , any updates on this issue? do you have further questions? or it's good to close this?

**Comment by lalalune** [comment]
Hi @SteveJunGao --

Got a bit busy but yes, still working on this.

Shapenet categories serve as okay labels, but we'd love nice descriptive video game-y embeddings if possible. I'm experimenting with captioning from BLIP (https://huggingface.co/spaces/Salesforce/BLIP) on various weapons and objects and it is very good on single objcts. LAION just did a similar thing with synthetic captioned dataset, to good result, so we will make a pipeline to add captions to our shapenet data and try to pass that along in training, appended to the class.

If you did post this your work with the NADA approach then I would be hugely grateful, but understand you are probably busy and it is not priority, so I will try to work on it as fast as I can :D

**Comment by benquick123** [comment]
Hi, are you still planning to release the CLIP conditioning code?
